Documentation/General/productPlanning/requirements.tex

```diff
@@ -0,0 +1,146 @@
+
+\item The web interface shall show a screen to log in to the application.
+
+\item The login mechanism shall be managed by SoundCloud OAuth keys ("Log in with SoundCloud").
+
+\item The login mechanism shall show an error page when the user does not authorize MoodCat to have access to their SoundCloud data.
+
+\item The web interface shall show a first-use tutorial upon logging in succesfully for the first time.
+
+\item The web interface shall show the user a page where they can select their current mood.
+
+\item The web interface shall show the following moods:
+		\begin{itemize}
+		\item Nervous
+		\item Exiting
+		\item Happy
+		\item Pleasing
+		\item Relaxing
+		\item Peaceful
+		\item Calm
+		\item Sleepy
+		\item Bored
+		\item Sad
+		\end{itemize}
+	  These moods are defined by the Arousal/Valence model\footnote{Y. Chang and H.H. Chen. Music Emotion Recognition. Boca raton, State:
+CRC Press, 2011.}.
+
+\item The system shall serve the user a list of rooms, ordered by how close they match the given mood.
+
+\item The web interface shall present the user with a room when it is chosen.
+	
+\item The system shall stream the music from the SoundCloud platform using the SoundCloud API\footnote{\url{https://developers.soundcloud.com/docs/api/guide}}
+
+\item The system shall use a multi-layer perceptron system to predict the mood of a song.
+
+\item The system will use a recommender system to determine the next song to be played in each room.
+
+\end{itemize}
+\subsubsection{Should haves}
+\begin{itemize}
+\item The web interface shall start playing the current song on the current timestamp when a user joins the room.
+
+\item A room shall contain a chat, connecting people with the same mood together.
+
+\item A room shall contain a now playing slider.
+
+\item The web interface shall give the user the option to mute the playing music.
+
+\item The web interface shall notify the user if they made an illogical combination of moods, warning them results may vary.
+
+\item A room has a button to upvote the current song, indicating the song is a good fit to the room.
+
+\item A room has a button to downvote the current song, indicating the song is a bad fit to the room. This will also vote to skip.
+
+\item A room has a button to "vote to skip". This does not affect the fit of the song to the room, but is a tool to skip songs.
+
+\item A song shall be skipped in one of the following situations:
+	\begin{itemize}
+	\item The song received $\frac{\$number\_of\_users\_in\_room}{3}$ votes to skip.
+	\item The song received at least 10 votes, and the ratio upvote:downvote is 1:2 or less.
+	\item A moderator skips the song (to prevent escalations)
+	\end{itemize}
+	
+\item The system shall periodically suggest the user to hop to another room to encourage more interaction and we can direct the user to a more preferred mood.
+
+\item The multi-layer perceptron shall be trained with metadata provided by EEMCS
```

Comment:
It will be trained by the users. The data is not from EEMCS (Friso comments). I would go for

> The systems cold start will be enabled by generating initial values from the metadata in our dataset.